# The Microbe Directory v2.0
*The ultimate microbe database*

Microbes include all unicellular organisms and so are extremely rich and diverse.


**The Microbe Directory (TMD)** is a collective research effort to profile and annotate more than 30,000 microbial species that include Bacteria, Archaea, Virus, Fungi, and Algae.

TMD aims to:

>* Provide a curated list of microbes from three kingdoms: Archaea, Bacteria and Eukarya
>* Compile microbial data from different [databases](#first) and [studies](#second) into a single one
>* Give a phenotypic and ecologic description of microbial species [features](#third).
>* Make microbial data handy to everyone!

---

## <a name="first">Databases</a>

* [Global Biodiversity Information Facility- GBIF](https://www.gbif.org/) an international network aimed to provide open access to data about all types of life on Earth. 
* [Virus-Host DB](https://www.genome.jp/virushostdb) a database about the relationships between viruses and their hosts.
* [HaloDom](http://halodom.bio.auth.gr) a database for halophilic organisms.
* [The Mycology Collections data Portal- MyCoPortal](http://mycoportal.org/portal/index.php) a network of universities, botanical gardens, museums, and agencies that provide taxonomic, environmental, and specimen-based information of Fungi.
* [BioCyc Database Collection](https://biocyc.org) a collection of Pathway/Genome Databases for Bacteria.
* [ISHAM Barcoding Database](http://its.mycologylab.org) a database for human/animal pathogenic fungal species.
* [FungiDB](https://fungidb.org/fungidb/) an integrated genomic and functional genomic database for Fungi. Includes experimental and environmental isolate sequence data.
* [RefSoil](https://www.nature.com/articles/ismej2016168#Sec6): A soil microbiome database

---

## <b name="second">Studies</b>

* [MetaSUB](http://metasub.org): Molecular profile of cities around the globe to improve their design, functionality, and impact on health. 
* [Earth Microbiome Project](http://www.earthmicrobiome.org): Characterization of microbial communities around the globe.
* [TARA Oceans](http://ocean-microbiome.embl.de/companion.html):  Metagenomic study of oceans samples in epipelagic and mesopelagic waters across the globe.
* [Soil bacterial and fungal communities across a pH gradient in an arable soil](https://qiita.ucsd.edu/study/description/94): Soils collected across a long-term liming experiment (pH 4.0-8.3).
* [The ecology of the phyllosphere](https://qiita.ucsd.edu/study/description/396): Bacterial communities from leaves of 56 tree species in Boulder, Colorado, USA.
* [Characterization of Airborne Microbial Communities at a High-Elevation Site and Their Potential To Act as Atmospheric Ice Nuclei](http://dx.doi.org/10.1128/AEM.00447-09): Atmospheric microbial abundance, community composition, and ice nucleation at a high-elevation site in northwestern Colorado.
* [Microbial community composition in a lowland tropical rain forest- Costa Rica](https://doi.org/10.1016/j.soilbio.2010.08.011): Plot-scale manipulations of organic matter inputs to soils correlate with shifts in microbial community composition in a lowland tropical rain forest.
* [Microbial communities on money](https://qiita.ucsd.edu/study/description/375)


---

## <x name="third">Features</x>

Since microbial communities are phenotypic and genetically diverse, microbial features are stipulated by domain.

#### Virus

1. Genetic material: Virus have either RNA or DNA as their genetic material
2. Strand: The nucleic acid may be single (*ss*) or double stranded (*ds*).
3. Capsid symmetry: The way in which the capsid units are arranged.
  * Helical 
  * Icosahedral
  * Complex
4. Envelop: The outer layer of a virus that protects the nucleic acid. Virus without envelop are called naked.
5. Is it a pathogen? If yes, which is its host.
   * Human
   * Animal
   * Plant

#### Bacteria and Archaea

1. Gram stain: Used to distinguish and classify bacterial species into two large groups: Gram-positive and Gram-negative. 
2. Antimicrobial resistance (AMR): Antimicrobial resistance occurs naturally over time, usually through genetic changes. However, the misuse and overuse of antimicrobials is accelerating this process. 
3. Type of metabolisms:  the nutrition mode of microbes according to the sources of energy and carbon needed for living, growth and reproduction. All sorts of combinations may exist in nature.
   * Primary source of energy:
     * Phototrophs: Light is absorbed in photo receptors and transformed into chemical energy
     * Chemotrophs: Bond energy is released from a chemical compound.
   * Primary sources of reducing equivalents:
     * Organotrophs: Organic compounds are used as electron donor.
     * Lithotrophs: Inorganic compounds are used as electron donor.
   * Primary sources of carbon
     * Heterotrophs: Organic compounds are metabolized to get carbon for growth and development.
     * Autotrophs: Carbon dioxide (CO2) is used as source of carbon.


#### Bacteria, Archaea and Eukarya

1. Biofilm forming: Biofilms are multicellular communities held together by a self-produced extracellular matrix.  Biofilms impact humans in many ways as they can form in natural, medical, and industrial settings.
2. Spore forming: Also referred to as endospores, are the dormant form of vegetative microbes and are highly resistant to physical and chemical influences.
3. Microbiome: Host or environment where microbes are usually found.
   * Host: Microbes might be commensal or pathogenic to their host. Commensal microbes are found to be crucial to the survival of their hosts.
     * Sponges
     * Corals
     * Fungi
     * Plant
     * Animal
     * Human: Body sites of [Human Microbiome Project](https://www.hmpdacc.org/hmp/) 
   * Soil: Microbes are essential for soils. They are main drivers of nutrient cycles in soils, decompose organic matter, promote plant growth and control pests and diseases. 
     * Tundra
     * Grassland
     * Croplands
     * Forest
       * Tropical
       * Temperate
       * Boreal
   * Extreme: Microbes that live in habitats considered hard to survive in due to its extreme conditions such as temperature, accessibility to different energy sources or under high pressure.
     * Desert
     * Polar
     * Deep ocean
     * Space
   * Water: Water can support the growth of many types of microorganisms. Microbes are main drivers of biogeochemical processes and nutrient cycling. 
     * Ocean
     * Fresh
     * Mangrove
     * Sediments
4. Is it a pathogen? if Yes, which is its host:
     * Sponges
     * Corals
     * Fungi
     * Plant
     * Animal
     * Human: Body sites of [Human Microbiome Project](https://www.hmpdacc.org/hmp/) 
5. Extremophile: a microbe that thrives in physically or geochemically extreme conditions that are detrimental to most life on Earth. Microbes that can only live under optimal conditions are called Mesophiles.
6. If extremophile, which type.
   * Acidophile: Microbes that live in acidic systems with pH -0.06 to 4.0.
   * Alkaliphile:  Microbes capable of survival in alkaline environments with pH 8.5–11 
   * Halophile:  Microbes that thrive in high salt concentrations. 
   * Metallotolerant: Microbes that survive in environments with a high concentration of dissolved heavy metals in solution
   * Barophile: Also called piezophile, are microbes which thrive at high pressures such as deep seas.
   * Psychrophile: Also called cryophiles, are microbes capable of growth in low temperatures, ranging from −20°C to 10°C.
   * Radioresistant: Microbes capable of withstand high levels of ionizing radiation.
   * Thermophile: Microbes that live at high temperatures between 41°C and 122°C.
   * Xerophile: Microbes that grow and reproduce in conditions with a low availability of water.
   * Hypolith: Organisms that live underneath rocks in cold deserts.
   * Oligotroph: Microbes capable of growth in nutritionally limited environments.



## Installation	

From source	
```	
git clone https://github.com/dcdanko/MD2	
cd MD2	
python setup.py install	
```

## Building NCBI taxonomic table
Command line usage	
```	
microbe_directory filter-microbes <output-file>	
```
The following outputs the taxonomy of all available Bacteria and Viruses from the NCBI dmp files. The table consists of the scientific name and classification from phylum-species level along with the unique taxonomic id.

## Merging NCBI files with all given datasets
Command line usage	
```	
MD2 annotate-taxa <file-path>	
```
The following outputs individual files for Bacteria, Viruses and Fungi merging NCBI unique taxonomic id, scientific name and the rank with the above mentioned databases. The file path for the following should be a folder containing all the databases intended to be merged. The dataset folder contains the merged databases.

## Cleaning the file
Command line usage	
```	
microbe_directory clean-file <input-file> <output-file>	
```
The input file consists of the file generated from previous step. The output contains clean, merged file dropping unimportant columns and rows. 

## Update Bacteria Columns
The Spore formation and Gram Negative columns are updated. Also, OTU tables are updated filling up Genus->Species. 

```
microbe_directory update-bacteria <input-file> <output-file>
```

### Dataset Creation with OTU File
If you have Biom File, the reference OTU file and metadata file, you can generate a table to reflect the various studies/feature. All input should be in csv and the biom file can be converted to csv using QIIME previously.

```
microbe_directory dataset-preprocessing [options] <otu-file> <biom-file> <metadata-file> <output-file>
```

### Dataset Creation for WGS File
When proper metadata is present, species file for desired feature can be obtained as follows.

```
microbe_directory metasub-preprocessing [options] <file> <metadata-file> <output-file>
```

### Column Statistics
The column statistics for before and after file can be obtained by the following.

```
microbe_directory column-compare <before-file> <after-file> <output-file>
```

### Row Statistics
How many columns for a particular species is filled.

```
microbe_directory stats-file <options> <file> <output-file>
```
